### ISSUE TITLE
CycleCL

### ISSUE THREAD

**Issue by guarin** [lightly-ai/lightly#1447]
CycleCL is a new contrastive learning method for cyclic videos: https://arxiv.org/abs/2311.03402

It could be an interesting addition to the package. We would have to add the following:
- [ ] Loading batches from a single video (ordered!)
- [ ] Finding negative and positive examples in a batch
- [ ] Loss -> It is basically Ntxentloss but generalized to accepting custom negative/positive pairs.
- [ ] Augmentations

**Comment by philippmwirth** [comment]
There already exists a triplet loss implementation in [torch](https://pytorch.org/docs/stable/generated/torch.nn.TripletMarginLoss.html). The main contribution would be to implement a triplet sampler that works on multiple GPUs and then a thin wrapper to pass the triplets to the triplet loss.

**Comment by philippmwirth** [comment]
We'd also need to add a custom projection head (they use 3D convolution).

**Comment by guarin** [comment]
Note: We are not planning on adding this for now.
